rtloader/three/three.cpp

```diff
@@ -60,45 +63,32 @@ Three::~Three()
 
 void Three::initPythonHome(const char *pythonHome)
 {
-    // Py_SetPythonHome stores a pointer to the string we pass to it, so we must keep it in memory
-    wchar_t *oldPythonHome = _pythonHome;
-    if (pythonHome == NULL || strlen(pythonHome) == 0) {
-        _pythonHome = Py_DecodeLocale(_defaultPythonHome, NULL);
-    } else {
-        _pythonHome = Py_DecodeLocale(pythonHome, NULL);
+    const auto home = pythonHome != nullptr && strlen(pythonHome) > 0 ? pythonHome : _defaultPythonHome;
+    const auto status = PyConfig_SetBytesString(&_config, &_config.home, home);
+    if (PyStatus_Exception(status)) {
+        setError("Failed to set python home");
```

Comment:
I don't believe anything will check for errors set this way when `initPythonHome` is called.

Reply:
Indeed not. I suppose the most appropriate way would be to check for `_error` in `Three::init`?

Reply:
Since all this does is setting values in the config now, maybe we can move this to `init`, after the pre-init phase?